### PR TITLE
chore(support-bundle): refactor out support bundle pkg

### PR DESCRIPTION
**Pull request by DanStough**
Refactoring to support using `support-bundle` as a package within kotsadm. Still work to do on consolidating things with what's in preflights but this is a first step.

### CODE REVIEW

File: cmd/troubleshoot/cli/run.go [hunk 1]
```diff
@@ -678,134 +372,50 @@ func getExpectedContentType(uploadURL string) string {
...
-	}
-
-	return nil
-}
-
-type CollectorFailure struct {
-	Collector *troubleshootv1beta2.Collect
-	Failure   string
-}
-
```
Comment (DanStough) [inline cmd/troubleshoot/cli/run.go:780]: As far as I can tell this isn't used in troubleshoot or kots

File: pkg/supportbundle/supportbundle.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+	var analyzeResults []*analyzer.AnalyzeResult
+
+	if len(spec.Analyzers) > 0 {
+
+		tmpDir, err := ioutil.TempDir("", "troubleshoot")
+		if err != nil {
+			return analyzeResults, errors.Wrap(err, "failed to make directory for analysis")
+		}
+		defer os.RemoveAll(tmpDir)
```
Comment (DanStough) [inline pkg/supportbundle/supportbundle.go:197]: new

File: pkg/supportbundle/supportbundle.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+		f, err := os.Open(archivePath)
+		if err != nil {
+			return analyzeResults, errors.Wrap(err, "failed to open support bundle for analysis")
+		}
+		defer f.Close()
```
Comment (DanStough) [inline pkg/supportbundle/supportbundle.go:203]: new

File: cmd/troubleshoot/cli/run.go (hunk 1 above)
Comment (divolgin) [reply cmd/troubleshoot/cli/run.go:780]: indeed

File: cmd/troubleshoot/cli/run.go [hunk 4]
```diff
@@ -147,71 +137,50 @@ func runTroubleshoot(v *viper.Viper, arg string) error {
...
+		ProgressChan:              progressChan,
+		SinceTime:                 sinceTime,
+	}
+
+	archivePath, err := supportbundle.CollectSupportBundleFromSpec(&supportBundle.Spec, additionalRedactors, createOpts)
 	if err != nil {
 		return errors.Wrap(err, "run collectors")
 	}
 
 	fmt.Printf("\r%s\r", cursor.ClearEntireLine())
```
Comment (divolgin) [inline cmd/troubleshoot/cli/run.go]: should this print to `c` instead of using the fmt package directly?

**Review by divolgin — APPROVED** [review]

File: cmd/troubleshoot/cli/run.go [hunk 5]
```diff
@@ -55,15 +56,15 @@ func runTroubleshoot(v *viper.Viper, arg string) error {
 		httpClient = http.DefaultClient
 	}
 
-	collectorContent, err := loadSupportBundleSpec(v, arg)
+	collectorContent, err := supportbundle.LoadSupportBundleSpec(arg, httpClient)
```
Comment (divolgin) [inline cmd/troubleshoot/cli/run.go]: I find it odd that httpClient is a global variable, but is passed around as a one off argument for some functions.  How about making something similar to `k8sutil.GetRESTConfig()` but for http client?

Comment (DanStough) [reply cmd/troubleshoot/cli/run.go]: Yes, that's a great point.

File: cmd/troubleshoot/cli/run.go (hunk 4 above)
Comment (DanStough) [reply cmd/troubleshoot/cli/run.go]: Not sure - this was existing. Proposing this instead?
```golang
c := color.New()
c.Println("\r%s\r", cursor.ClearEntireLine())
```

Comment (divolgin) [reply cmd/troubleshoot/cli/run.go]: yeah, seems like an oversight from before.  not related to your changes.

File: pkg/httputil/config.go [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func AddTransport(transport *http.Transport) {
+	httpTransport = transport
+}
+
+func GetHttpClient() *http.Client {
+
+	if httpTransport != nil {
+		client := &http.Client{Transport: httpTransport}
```
Comment (divolgin) [inline pkg/httputil/config.go]: This will leak TCP sockets.  Should reuse the same client, similar to `http.DefaultClient`

**Review by divolgin — APPROVED** [review]

**Pull request merged**